### PR TITLE
chore: update readme code example to new API.

### PR DESCRIPTION
**Pull request by saul-avikar**
This PR simply fixes the code example provided by the readme.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -55,7 +56,7 @@ gsub.publish('fruit', new TextEncoder().encode('banana'))
 
 ```js
 const options = {…}
-const gossipsub = new Gossipsub(libp2p, options)
+const gossipsub = gossipsub(options)(libp2p)
```
Comment (mpetrunic) [inline README.md:59]: maybe makes more sense to put example where to pass `gossipsub(options)` in libp2p init

**Review by mpetrunic — COMMENTED** [review]
Could you sign CLA agreement^^

**Comment by saul-avikar** [comment]
> Please sign CLA agreement

I don't believe these CLAs are healthy to the opensource ecosystems since they are non-standard and contain legalese that only someone with good understanding of it can fully understand the extent of its meaning. Therefore I will not be signing it, however someone who has the ability to make the changes should update the readme to reflect the new API in a similar fashion to what I have done.

**Comment by wemeetagain** [comment]
Not sure why we have the CLA here since this is a permissively licensed repo :thinking:

**Comment by mpetrunic** [comment]
> Not sure why we have the CLA here since this is a permissively licensed repo :thinking: 
> 

I think because of that ethereum c++ client that couldn't change license as external contributors were claiming that they only submitted code under previous license so CLA forces everyone to give out any rights to submitted code to chainsafe

**Comment by wemeetagain** [comment]
Shouldn't that only be an issue when code isn't permissively licensed?

The c++ ethereum client was GPL licensed.

**Comment by philknows** [comment]
> > Please sign CLA agreement
> 
> I don't believe these CLAs are healthy to the opensource ecosystems since they are non-standard and contain legalese that only someone with good understanding of it can fully understand the extent of its meaning. Therefore I will not be signing it, however someone who has the ability to make the changes should update the readme to reflect the new API in a similar fashion to what I have done.

We've agreed that CLA agreements for Lodestar should not exist on libraries such as this one, but only retained for our main Lodestar monorepo. @q9f is in the process of helping us to remove this requirement and hopefully that will allow this PR to be merged if there's no additional issues.

**Comment by q9f** [comment]
You can ignore CLA here. It's not meant for the libraries.

**Review by wemeetagain — APPROVED** [review]

**Pull request merged**